gha-cache/src/api.rs

```diff
@@ -380,6 +408,14 @@ impl Api {
 
                     drop(permit);
 
+                    if let Err(Error::ApiError {
```

Comment:
Do you want to use the `circuit_breaker_trip_on_429` function here, so that we don't have to implement this twice (and also miss out on the `info` log that is present in that function but not here?)

Reply:
Unfortunately that is on `self`, and I can't refer to self here :(.

Reply:
Could you make it a free-standing function that only takes in the error and the AtomicBool, so it doesn't need `self` anymore?

Reply:
Nice idea, done :).